homeassistant/components/esphome/voice_assistant.py

```diff
@@ -177,19 +186,119 @@ def _event_callback(self, event: PipelineEvent) -> None:
                 "code": event.data["code"],
                 "message": event.data["message"],
             }
-            self.handle_finished()
+            self._tts_done.set()
+            error = True
 
         self.handle_event(event_type, data_to_send)
+        if error:
+            self.handle_finished()
+
+    async def _wait_for_speech(
+        self,
+        segmenter: VoiceCommandSegmenter,
+        chunk_buffer: MutableSequence[bytes],
+    ) -> bool:
+        """Buffer audio chunks until speech is detected.
```

Comment:
You’re not buffering here, but throwing them away.
```suggestion
        """Discard audio chunks until speech is detected.
```

Reply:
Oh never mind, missed the buffer. 

So uh, then I guess: do we need to buffer audio chunks if it’s silence?

Reply:
I followed the voip code, but yeah probably not tbh.

Reply:
It's important to have some buffer because VAD isn't necessarily instantaneous (single chunk). It may take several chunks to reach a threshold, so having the last few "silence" chunks will avoid cutting off part of the first word.